subprojects/internal-performance-testing/src/main/groovy/org/gradle/performance/results/IndexPageGenerator.java

```diff
@@ -76,11 +76,11 @@ public IndexPageGenerator(ResultsStore resultsStore, File resultJson) {
     private ScenarioBuildResultData queryExecutionData(ScenarioBuildResultData scenario) {
         PerformanceTestHistory history = resultsStore.getTestResults(scenario.getScenarioName(), DEFAULT_RETRY_COUNT, PERFORMANCE_DATE_RETRIEVE_DAYS, ResultsStoreHelper.determineChannel());
```

Comment:
I think we need to query for more than `DEFAULT_RETRY_COUNT` results, given that the build id we search for may not be one the last ones.

Reply:
I would go for `3 * DEFAULT_RETRY_COUNT`, that would allow three instances of the test which ran in parallel and retried.

If `currentCommitExecutions` is empty, then I would should only the `DEFAULT_RETRY_COUNT` entries as we do now.

BTW: Do we need to do anything for the cached scenarios? With the current code we won't be able to match them up, right? That can be done a follow up commit.